refactor(actions): use `Omit` to avoid leaking types to shared context (#13429)
withastro/astro: .changeset/calm-beans-roll.md

---
'astro': patch
---

> The methods `ActionAPIContext.rewrite` has been marked as deprecated and it will be removed in the future major versions of Astro.

Reviewer: ```suggestion
The `ActionAPIContext.rewrite` methof is deprecated and will be removed in the future major version of Astro
```

Reviewer: I updated my suggestion
